src/measure_theory/measurable_space.lean

```diff
@@ -267,6 +267,15 @@ begin
   exact (hf ht).inter h.measurable_set.of_compl,
 end
 
+omit mβ
+
+@[measurability] lemma measurable_set_preimage_iterate {f : α → α} (hf : measurable f)
+  (hs : measurable_set s) (n : ℕ) :
+  measurable_set $ (preimage f)^[n] s :=
+begin
+  induction n with n ih, { simp [hs], },
+  simpa only [iterate_succ', comp_app] using hf ih,
+end
```

Comment:
It feels wrong to need an induction by this point. Shouldn't the proof rather be "`(preimage f)^[n] = preimage (f^[n])` and `f^[n]` is measurable"?

Reply:
Indeed I now note that `measurable.iterate` would be the right thing to prove here but it already exists so I'm dropping this lemma.